tracking/include/pcl/tracking/impl/particle_filter.hpp

```diff
@@ -379,11 +379,11 @@ ParticleFilterTracker<PointInT, StateT>::update()
   StateT orig_representative = representative_state_;
   representative_state_.zero();
   representative_state_.weight = 0.0;
-  for (std::size_t i = 0; i < particles_->points.size(); i++) {
-    StateT p = (*particles_)[i];
+  for (const auto& point : *particles_) {
+    StateT p = point;
```

Comment:
If you're going to make a copy, it would be better to make it already inside the for-range, request `const auto` instead of `const auto&`